### ISSUE TITLE
Unrecognized option '--projectid'

### ISSUE THREAD

**Issue by hades200082** [Kentico/cloud-generators-net#45]
`--ProjectID` and `-p` both work but `--projectid` gives the error "Unrecognized option '--projectid'"

**Comment by petrsvihlik** [comment]
Confirmed, it's a bug.

**Comment by kashifsoofi** [comment]
A CommandOption can be resolved with 3 names, symbol, short name or long name e.g. "-?|-h|--help", this can be resolved by treating the all small long name as short name so would have to be replaced with -projectid. It can be a work around and this cannot be termed as a bug (only a limitation) as its not something supported by command line parser in use.

**Comment by hades200082** [comment]
--ProjectID and --projectid should be interchangeable.  Can this not be handled by forcing a lowercase comparison somewhere? (not got the code in front of my right now)

**Comment by petrsvihlik** [comment]
The parsing's done [here](https://github.com/natemcmaster/CommandLineUtils/blob/master/src/CommandLineUtils/CommandOption.cs#L27)

Ideally, I'd like to preserve both --ProjectID and --projectid. The pascal case is necessary for the [configuration](https://github.com/Kentico/cloud-generators-net/blob/master/src/CloudModelGenerator/Program.cs#L30) to work correctly (afaik).

So either we find a way to handle this by lowercasing it at some point or we replace the parser. With [System.CommandLine](https://github.com/Kentico/cloud-generators-net/issues/46) for instance.

**Comment by hades200082** [comment]
```
if (part.StartsWith("--"))
                {
                    LongName = part.Substring(2);
                }
```

Add `.ToLower()` on the end of `part.Substring(2)` then just ensure that everywhere you check it you use only the lowercase variant.

**Comment by petrsvihlik** [comment]
The code is part of a referenced nuget package...we won't be changing that.

**Comment by kashifsoofi** [comment]
As you can see CommandLineUtils.CommandOption would only take the last option name, in our case 2nd -- in template would overwrite the -- lower case option in LongName. Both options with all lowercase or PascalCase cannot be used with current referenced package, so in my opinion its not a bug rather limitation of current parser. Another hack can be to have another option with all lowercase.

**Comment by petrsvihlik** [comment]
It actually is a bug because it worked before we introduced the PascalCase option.

Hmm...interesting idea with the second options...but I'd rather avoid hacks. Let's come up with a solid design and solve it once and for all :)

**Comment by petrsvihlik** [comment]
fixed by @kashifsoofi  in #47
